ccm_web/client/src/components/GradebookCanvasValidators.ts

```diff
@@ -1,13 +1,32 @@
 import { GradebookRecord } from '../pages/GradebookCanvas'
 
+enum GradbookRowInvalidationType {
+  ERROR,
+  WARNING
+}
+
 interface GradebookRowInvalidation {
   message: string
   rowNumber: number
+  type: GradbookRowInvalidationType
+}
+class CurrentAndFinalGradeMismatchError implements GradebookRowInvalidation {
+  message: string
+  record: GradebookRecord
+  rowNumber: number
+  type: GradbookRowInvalidationType = GradbookRowInvalidationType.ERROR
+  constructor (record: GradebookRecord, rowNumber: number) {
+    this.record = record
+    this.rowNumber = rowNumber
+    this.message = 'Current and Final grade mismatch: ' + record.Student + '(' + record['SIS Login ID'] + ')'
+  }
 }
-class CurrentAndFinalGradeMismatchInvalidation implements GradebookRowInvalidation {
+
+class CurrentAndFinalGradeMismatchWarning implements GradebookRowInvalidation {
```

Comment:
Both `CurrentAndFinalGradeMismatchError` and `CurrentAndFinalGradeMismatchWarning ` class just vary by one variable `type: GradbookRowInvalidationType`. I feel this can still be single class taking `GradbookRowInvalidationType` error or warning as parameter.

Reply:
pseudo code
```
class CurrentAndFinalGradeMismatchHandler implements GradebookRowInvalidation {
  message: string
  record: GradebookRecord
  rowNumber: number
  type: GradbookRowInvalidationType = GradbookRowInvalidationType.ERROR
  constructor (record: GradebookRecord, rowNumber: number, type: enum) {
    this.record = record
    this.rowNumber = rowNumber
    this.message = 'Current and Final grade mismatch: ' + record.Student + '(' + record['SIS Login ID'] + ')'
    this.type = type // Error or Warning depending on class instantiation
 
  }
}
```

Does this work?

Reply:
I knew someone would say this lol.. I didn't think it added value, and might have to change if the enum was extended for some reason.  Anyways I made a change, though I didn't do it the way you suggested.